src/types/deep.select.properties.ts

```
@@ -0,0 +1,27 @@
+export type DeepSelectProperty<Property> =
+  Property extends Promise<infer I>
+    ? DeepSelectProperty<I> | boolean
+    : Property extends Array<infer I>
+      ? DeepSelectProperty<I> | boolean
+      : Property extends string
+        ? boolean
+        : Property extends number
+          ? boolean
+          : Property extends boolean
+            ? boolean
+            : Property extends Function
```

Biome

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)